strawberryfields/gbs/plot.py

```diff
@@ -0,0 +1,235 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Plotting and visualization
+==========================
+
+**Module name:** :mod:`strawberryfields.gbs.plot`
+
+.. currentmodule:: strawberryfields.gbs.plot
+
+This module provides functionality for visualizing GBS outputs as graphs and point processes.
+
+.. autosummary::
+    _node_coords
+    edge_coords
+    plot
+    plot_subgraph
+
+Code details
+^^^^^^^^^^^^
+"""
+from typing import Tuple, Optional
+
+import networkx as nx
+import plotly.graph_objects as go
+
+
+def _node_coords(graph: nx.Graph, l: dict) -> Tuple:
+    """ Provides the coordinates for the graph nodes when given an input graph layout.
+
+    Args:
+        graph (nx.Graph): input graph
+        l (dict): dictionary of nodes and their respective coordinates
+
+    Returns:
+         Tuple: x and y coordinates for each node
+    """
+    n_x = []
+    n_y = []
+
+    for n in graph.nodes():
+        n_x.append(l[n][0])
+        n_y.append(l[n][1])
+
+    return {"x": n_x, "y": n_y}
+
+
+def edge_coords(graph: nx.Graph, l: dict) -> dict:
+    """ Provides the coordinates for the graph edges when given an input graph layout.
+
+        Args:
+            graph (nx.Graph): input graph
+            l (dict): dictionary of edges and their respective coordinates
+
+        Returns:
+             dict: x and y coordinates for beginning and end of each edge
+        """
+    e_x = []
+    e_y = []
+
+    for e in graph.edges():
+
+        start_x, start_y = l[e[0]]
+        end_x, end_y = l[e[1]]
+
+        e_x.append(start_x)
+        e_x.append(end_x)
+
+        e_y.append(start_y)
+        e_y.append(end_y)
+
+        e_x.append(None)
+        e_y.append(None)
+
+    return {"x": e_x, "y": e_y}
+
+
+web_green1 = '#84ba5b'
+web_green_light1 = '#e0edd6'
+web_green2 = '#3e9651'
+web_red1 = '#d35e60'
+web_red2 = '#cc2529'
+sf_cyan = '#3e9bb7'
+sf_cyan_light = '#6eb4c9'
+sf_lightgreen = '#daf0e9'
+sf_red = '#d35e60'
+sf_grey = '#efefef'
+light_grey = '#CDCDCD'
+
+graph_node_colour = web_green2
+graph_edge_colour = light_grey
+subgraph_node_colour = web_red2
+subgraph_edge_colour = web_red2
+
+graph_node_size = 14
```

Comment:
The things we do keep, they can be all capitalized to follow the module level constant style: https://www.programiz.com/python-programming/variables-constants-literals